Source/WebCore/html/HTMLElement.cpp

```diff
@@ -1217,14 +1222,29 @@ ExceptionOr<void> HTMLElement::hidePopover()
     return hidePopoverInternal(FocusPreviousElement::Yes, FireEvents::Yes);
 }
 
-ExceptionOr<bool> HTMLElement::togglePopover(std::optional<bool> force)
+ExceptionOr<bool> HTMLElement::togglePopover(std::optional<std::variant<WebCore::HTMLElement::TogglePopoverOptions, bool>> options)
 {
+    std::optional<bool> force;
+    HTMLElement* invoker = nullptr;
+
+    if (options.has_value()) {
+        WTF::switchOn(options.value(),
+            [&force, &invoker](TogglePopoverOptions options) {
+                force = options.force;
+                invoker = options.source.get();
+            },
+            [&force](bool value) {
```

Comment:
```suggestion
            [&](TogglePopoverOptions options) {
                force = options.force;
                invoker = options.source.get();
            },
            [&](bool value) {
```
This is slightly more elegant if it works.